views/js/front/hosted-templates/hosted_fields.js

```diff
@@ -30,15 +30,6 @@ $(document).ready(function () {
     $('[id="payment-form"]').on('submit', function (event) {
         event.preventDefault();
 
-        var paymentType = $(this).find("[name=saferpayPaymentType]").val();
-
-        //NOTE: if it's not a hosted iframe then we don't need to submitHostedFields.
-        if (paymentType !== saferpay_payment_types.hosted_iframe) {
```

Comment:
To always use PaymentPage with already saved cards. For less complexity with redirecting